### PR TITLE
[SYCLomatic] Fix migration of cv-qualified cg::thread_block in function parameter

### PR DESCRIPTION
**Pull request by jzc**
Signed-off-by: Cai, Justin <justin.cai@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -2702,8 +2702,8 @@ void TypeInDeclRule::runRule(const MatchFinder::MatchResult &Result) {
       }
     }
 
-    if (DpctGlobalInfo::getTypeName(TL->getType().getCanonicalType()) ==
-        "cooperative_groups::__v1::thread_block") {
+    if (CanonicalTypeStr.find("cooperative_groups::__v1::thread_block")
```
Comment (0x12CC) [inline clang/lib/DPCT/ASTTraversal.cpp]: This condition will be true for `"cooperative_groups::__v1::thread_block_tile<"` from line 2680 when `ETL` is `nullptr`. Would it be better to explicitly remove the qualifiers and compare for equality?

Comment (jzc) [reply clang/lib/DPCT/ASTTraversal.cpp]: Yea, good point. I'll make this fix.

**Review by 0x12CC — APPROVED** [review]

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 2]
```diff
@@ -2673,8 +2673,8 @@ void TypeInDeclRule::runRule(const MatchFinder::MatchResult &Result) {
       EndLoc = SM->getExpansionRange(TL->getBeginLoc()).getEnd();
     }
 
-    std::string CanonicalTypeStr =
-        DpctGlobalInfo::getTypeName(TL->getType().getCanonicalType());
+    std::string CanonicalTypeStr = DpctGlobalInfo::getTypeName(
```
Comment (ziranzha) [inline clang/lib/DPCT/ASTTraversal.cpp]: How about use `DpctGlobalInfo::getUnqualifiedTypeName`?

**Review by ziranzha — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**